From 233218d45a88d0de97b1c486fa54ffe1f757fd8b Mon Sep 17 00:00:00 2001
From: Andrey Borzenkov <arvidjaar@gmail.com>
Date: Fri, 3 May 2013 18:39:17 +0400
Subject: [PATCH 413/482] Rename grub-core/tests/checksums.c into
 grub-core/tests/checksums.h and add it as source to functional_test module.

---
 ChangeLog                                    | 5 +++++
 grub-core/Makefile.core.def                  | 1 +
 grub-core/tests/{checksums.c => checksums.h} | 0
 grub-core/tests/video_checksum.c             | 4 ++--
 4 files changed, 8 insertions(+), 2 deletions(-)
 rename grub-core/tests/{checksums.c => checksums.h} (100%)

diff --git a/ChangeLog b/ChangeLog
index 1b4c925..8113b6d 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2013-05-03  Andrey Borzenkov <arvidjaar@gmail.com>
+
+	Rename grub-core/tests/checksums.c into grub-core/tests/checksums.h
+	and add it as source to functional_test module.
+
 2013-05-03  Vladimir Serbinenko  <phcoder@gmail.com>
 
 	* grub-core/tests/video_checksum.c: Don't set GENERATE_MODE.
diff --git a/grub-core/Makefile.core.def b/grub-core/Makefile.core.def
index da7637f..469524f 100644
--- a/grub-core/Makefile.core.def
+++ b/grub-core/Makefile.core.def
@@ -1731,6 +1731,7 @@ module = {
   name = functional_test;
   common = tests/lib/functional_test.c;
   common = tests/lib/test.c;
+  common = tests/checksums.h;
   common = tests/video_checksum.c;
   common = tests/fake_input.c;
   common = video/capture.c;
diff --git a/grub-core/tests/checksums.c b/grub-core/tests/checksums.h
similarity index 100%
rename from grub-core/tests/checksums.c
rename to grub-core/tests/checksums.h
diff --git a/grub-core/tests/video_checksum.c b/grub-core/tests/video_checksum.c
index 41a262a..f3eb9fc 100644
--- a/grub-core/tests/video_checksum.c
+++ b/grub-core/tests/video_checksum.c
@@ -613,7 +613,7 @@ struct checksum_desc
 };
 
 const struct checksum_desc checksum_table[] = {
-#include "checksums.c"
+#include "checksums.h"
 };
 
 void
@@ -625,7 +625,7 @@ grub_video_checksum (const char *basename_in)
 
 #if defined (GENERATE_MODE) && defined (GRUB_MACHINE_EMU)
   if (genfd < 0)
-    genfd = open ("checksums.c", O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
+    genfd = open ("checksums.h", O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
   if (genfd >= 0)
     {
       char buf[400];
-- 
1.8.2.1

